src/Data/Functor/Foldable/TH.hs

```diff
@@ -70,12 +74,59 @@ import Paths_recursion_schemes (version)
 -- as we don't try to do better than
 -- <https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#deriving-functor-instances GHC's DeriveFunctor>.
 --
-makeBaseFunctor :: Name -> DecsQ
-makeBaseFunctor = makeBaseFunctorWith baseRules
+-- Some doctests:
+--
+-- >>> data Expr a = Lit a | Add (Expr a) (Expr a) | Expr a :* [Expr a]
+-- >>> ; makeBaseFunctor ''Expr
```

Comment:
These doctests are nice. But unless I'm not mistaken, I don't think they're actually being run right now. We should either set up `doctests` properly or add test cases. (I'd be slightly in favor of the latter, but I don't feel strongly on this.)

Reply:
They are. `doctest: True` in `haskell-ci.cabal`. (and also `make doctest`).

I don't want to use `cabal-doctest`.

Reply:
Oops, I forgot that we use `haskell-ci`-provided `doctest`s in this repo. In that case, this is fine.